server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -1521,11 +1520,6 @@ public String getChannelType() {
             return "direct";
         }
 
-        @Override
-        public TransportVersion getVersion() {
```

Comment:
This is in DirectResponseChannel, which should always use `TransportVersion.CURRENT` as you're talking to the local node